src/main/java/dao/AccountDao.java

```diff
@@ -0,0 +1,87 @@
+package dao;
+
+import model.Account;
+import model.Card;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Optional;
+
+public record AccountDao(DBCreator dbCreator) {
+
+    private static final String GET_ACCOUNT_QUERY = "SELECT number, pin, balance FROM account WHERE number = ? AND pin = ?";
+    private static final String ACCOUNT_IS_PRESENT_QUERY = "SELECT number FROM account WHERE number = ?";
+    private static final String CREATE_ACCOUNT_QUERY = "INSERT INTO account (number, pin) VALUES (?, ?)";
+    private static final String UPDATE_ACCOUNT_QUERY = "UPDATE account SET balance = balance + ? WHERE number = ?";
+    private static final String DELETE_ACCOUNT_QUERY = "DELETE FROM account WHERE number = ?";
+
+    public Optional<Account> get(String cardNumber, String cardPIN) {
+        try (PreparedStatement statement = dbCreator.getConnection().prepareStatement(GET_ACCOUNT_QUERY)) {
+            statement.setString(1, cardNumber);
+            statement.setString(2, cardPIN);
+            ResultSet resultSet = statement.executeQuery();
+
+            int balance = resultSet.getInt("balance");
```

Comment:
The balance may not be present, can you initialize it after the `if statement`?

Reply:
fixed